File: app/src/main/java/com/example/footapp/ui/item/ItemAdapter.java
package com.example.footapp.ui.item;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.example.footapp.R;
import com.example.footapp.model.Item;

import java.util.List;

public class ItemAdapter extends RecyclerView.Adapter<ItemAdapter.ItemViewHolder> {

    private final List<Item> mListItem;
    private ItemInterface mItemInterface;

    public ItemAdapter(List<Item> mListItem, ItemInterface mItemInterface) {
        this.mListItem = mListItem;
        this.mItemInterface = mItemInterface;
    }

    @NonNull
    @Override
    public ItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_item, parent, false);
        return new ItemViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ItemViewHolder holder, int position) {
        Item item = mListItem.get(position);
        if (item == null) {
            return;
        }
        holder.tvItemName.setText(item.getName());
        holder.tvItemAmount.setText(String.valueOf(item.getAmount()));
        holder.tvItemPrice.setText(String.valueOf(item.getPrice()));

        Glide.with(holder.imgAvatar.getContext()).load(item.getImgUrl()).into(holder.imgAvatar);

        holder.imgUpdate.setOnClickListener(v -> {
            Intent intent = new Intent(holder.imgUpdate.getContext(), UpdateItem.class);
            Bundle bundle = new Bundle();
            bundle.putSerializable("item", item);
            intent.putExtras(bundle);
            holder.imgUpdate.getContext().startActivity(intent);
        });
        holder.imgDelete.setOnClickListener(v -> {
            mItemInterface.deleteItem(position, item);
        });

    }

    @Override
    public int getItemCount() {
        if (mListItem != null) {
            return mListItem.size();
        }
        return 0;
    }

    public static class ItemViewHolder extends RecyclerView.ViewHolder {
        private ImageView imgAvatar, imgUpdate, imgDelete;
        private TextView tvItemName, tvItemAmount, tvItemPrice;

        public ItemViewHolder(@NonNull View itemView) {
            super(itemView);

            imgAvatar = itemView.findViewById(R.id.img_item);
            imgUpdate = itemView.findViewById(R.id.img_update);
            imgDelete = itemView.findViewById(R.id.img_delete);
            tvItemName = itemView.findViewById(R.id.tv_item_name);
            tvItemAmount = itemView.findViewById(R.id.tv_item_amount);
            tvItemPrice = itemView.findViewById(R.id.tv_item_price);
        }
    }

    public void remove(int pos) {
        mListItem.remove(pos);
        notifyItemRemoved(pos);
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/example/footapp/ui/orderlist/ItemAdapter.kt
package com.example.footapp.ui.orderlist

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.footapp.ItemSize
import com.example.footapp.R
import com.example.footapp.databinding.ItemCatgoryBinding
import com.example.footapp.databinding.ItemDetailCatgoryBinding
import com.example.footapp.model.BillItem
import com.example.footapp.utils.formatToPrice

class ItemAdapter(val context: Context, var list: MutableList<BillItem>) :
    RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val binding = DataBindingUtil.inflate(
            layoutInflater,
            R.layout.item_detail_catgory,
            parent,
            false,
        ) as ItemDetailCatgoryBinding
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        (holder as ViewHolder).bind(list[position])
    }

    override fun getItemCount(): Int {
        return list.size
    }

    class ViewHolder(val binding: ItemDetailCatgoryBinding) :
        RecyclerView.ViewHolder(binding.root) {
        fun bind(item: BillItem) {
            if (item.item?.imgUrl?.isNotEmpty() == true) {
                Glide.with(binding.root.context)
                    .load(item.item.imgUrl!![0]).into(binding.ivProduct)
            } else {
                binding.ivProduct.setImageResource(R.drawable.ic_picture_nodata)
            }
            binding.tvNameProduct.text = item.item?.name
            binding.amount.text = item.quantity.toString()
            binding.tvPrice.text = item.price.formatToPrice()
            binding.amount.text = "x: ${item.quantity}"
            binding.tvSize.text = when (item.size) {
                ItemSize.S.ordinal -> ItemSize.S.name
                ItemSize.M.ordinal -> ItemSize.M.name
                ItemSize.L.ordinal -> ItemSize.L.name
                else -> ""
            }
            binding.tvDescription.text = item.description?:""
            println(item.description)
        }
    }
}
